services/server/port/createkind411.go

```diff
@@ -0,0 +1,230 @@
+		Software:      viper.GetString("RelaySoftware"),
+		Version:       viper.GetString("RelayVersion"),
+		DHTkey:        viper.GetString("RelayDHTkey"),
+	}
+
+	// Convert relay info to JSON
+	content, err := json.Marshal(relayInfo)
+	if err != nil {
+		return fmt.Errorf("error marshaling relay info: %v", err)
+	}
+
+	// Create the event
+	event, err := createAnyEvent(privateKey, publicKey, 411, string(content), nil)
+	if err != nil {
+		return fmt.Errorf("error creating kind 411 event: %v", err)
+	}
+
+	// Store the new event
+	if err := store.StoreEvent(event); err != nil {
+		return fmt.Errorf("error storing kind 411 event: %v", err)
+	}
+
+	// Print the event for verification
+	eventJSON, err := json.MarshalIndent(event, "", "  ")
+	if err != nil {
+		log.Printf("Error marshaling event for printing: %v", err)
+	} else {
+		log.Printf("Created and stored kind 411 event:\n%s", string(eventJSON))
+	}
+
+	log.Println("Kind 411 event created and stored successfully")
+	return nil
+}
+
+func createAnyEvent(privateKey *secp256k1.PrivateKey, publicKey *secp256k1.PublicKey, kind int, content string, tags []nostr.Tag) (*nostr.Event, error) {
+	stringKey := hex.EncodeToString(schnorr.SerializePubKey(publicKey))
+	log.Println("Public Key: ", stringKey)
+
+	event := &nostr.Event{
+		PubKey:    stringKey,
+		CreatedAt: nostr.Timestamp(time.Now().Unix()),
+		Kind:      kind,
+		Tags:      tags,
+		Content:   content,
+	}
+
+	serializedEvent := serializeEventForID(event)
+	hash := sha256.Sum256([]byte(serializedEvent))
+	eventID := hex.EncodeToString(hash[:])
+	event.ID = eventID
+
+	signature, err := schnorr.Sign(privateKey, hash[:])
+	if err != nil {
+		return nil, err
+	}
+
+	sigHex := hex.EncodeToString(signature.Serialize())
+	event.Sig = sigHex
+
+	err = signing.VerifySignature(signature, hash[:], publicKey)
+	if err != nil {
+		log.Printf("error verifying signature, %s", err)
+		return nil, fmt.Errorf("error verifying signature, %s", err)
+	} else {
+		log.Println("Signature is valid.")
+	}
+
+	return event, nil
+}
+
+func serializeEventForID(event *nostr.Event) string {
+	// Assuming the Tags and other fields are already correctly filled except ID and Sig
+	serialized, err := json.Marshal([]interface{}{
+		0,
+		event.PubKey,
+		event.CreatedAt,
+		event.Kind,
+		event.Tags,
+		event.Content,
+	})
+	if err != nil {
+		panic(err) // Handle error properly in real code
+	}
+
+	// Convert the JSON array to a string
+	compactSerialized := string(serialized)
+
+	return compactSerialized
+}
+
+func loadSecp256k1Keys() (*btcec.PrivateKey, *btcec.PublicKey, error) {
+
+	privateKey, publicKey, err := signing.DeserializePrivateKey(os.Getenv("NOSTR_PRIVATE_KEY"))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting keys: %s", err)
+	}
+
+	return privateKey, publicKey, nil
+}
+
+func generateEd25519Keypair(privateKeyHex string) (ed25519.PrivateKey, ed25519.PublicKey, error) {
```

Comment:
This may have been obsoleted by the client making the key now. Just guessing, tho.

Reply:
I will look into it as well. If it is not necessary I will remove it.